app/services/requisicion_service.py

```
@@ -0,0 +1,536 @@
+        # Si no es Admin/Superadmin, solo ve sus propias requisiciones o las de su departamento
+        # Esto es un filtro general, la lógica de `listar_requisiciones` es más específica para "activos"
+        if rol_nombre not in ['Admin', 'Superadmin']:
+            if hasattr(current_user, 'departamento_id') and current_user.departamento_id:
+                 query = query.filter(
+                     db.or_(
+                         Requisicion.creador_id == current_user.id,
+                         Requisicion.departamento_id == current_user.departamento_id # Ver las de su departamento
+                     )
+                 )
+            else:
+                query = query.filter_by(creador_id == current_user.id)
```

Ruff (0.11.9)

Undefined name `creador_id`

(F821)